devices/wholeBodyDynamics/UnknownWrenchContact.cpp

```diff
@@ -0,0 +1,64 @@
+#include "UnknownWrenchContact.h"
+
+namespace wholeBodyDynamics
+{
+
+UnknownWrenchContact::UnknownWrenchContact()
```

Comment:
I'm now confused. What's the difference between this class and http://wiki.icub.org/codyco/dox/html/idyntree/html/structiDynTree_1_1UnknownWrenchContact.html#a4b96dc6c0dad0f2d4cda49d65381b7d0 ? Why did we choose to define a new data structure for the same entity?

Reply:
@fjandrad @prashanthr05 I posted this reply but in the "files changed" page, I am assuming it didn't appear for you guys.

My idea was:
`wholeBodyDynamics::UnknownWrenchContact` inherits from `iDynTree::UnknownWrenchContact` and have in addition other variables and methods, they are:

```cxx
std::string frameName;  
uint8_t nrUnknownsInExtWrench;
constructor(); //it contains different parameters than what iDynTree::UnknownWrenchContact has
isTypeValid();
display();
```

Let me know if you suggest a better way to do it.

Reply:
The variable frameName is used only in `display()`, that I used only for debugging.

The key difference is the constructor of this class. It takes the `type` as string, the position and the direction as vectors of double, and then it creates the `iDynTree` objects (`UnknownWrenchContactType`, Position, and `Direction`). See:
```cxx
UnknownWrenchContact::UnknownWrenchContact(std::string parsedframeName, std::string parsedType, std::vector<double> parsedDirection, std::vector<double> parsedPosition)
```

Also `isTypeValid()`, that checks for the errors.

Reply:
I see. You probably discussed with @traversaro about it. My question is why couldn't this changes go directly to idyntree? It might be a desirable functionality for people who might use idyntree alone.

Reply:
True. IMO, it's better to open a PR against iDynTree to update the existing data structure rather than having two of them in two different places with almost the same name and only a minimal updated functionality (most of the checks done in this class can be avoided). That too, given the very minimal changes that this class introduces.

Also here we are introducing some vulnerabilities to cause the error.

Reply:
When we were first discussing this, I had in mind only filling the iDynTree UnknownWrench object appropriately with the use of existing iDynTree Wrench types. In this way, we reused efficiently the available code and reduced the amount of additional work we had to do.

@HosameldinMohamed  please correct me if I had understood wrongly?

Reply:
I think the main purpose of introducing this class, was to check the Yarp configuration parameters which are text-based and fill the corresponding iDynTree object which is enum class-based for wrench types and iDyntree position and direction based. 

I totally see this can be efficiently handled within the openContactFrames(config) function directly, instead of introducing this intermediate class. This way, it could be straight-forward.

Reply:
> When we were first discussing this, I had in mind only filling the iDynTree UnknownWrench object appropriately with the use of existing iDynTree Wrench types. In this way, we reused efficiently the available code and reduced the amount of additional work we had to do.
> 
> @HosameldinMohamed please correct me if I had understood wrongly?

@prashanthr05 No that was the idea, but then as you said :point_down: 

> I think the main purpose of introducing this class, was to check the Yarp configuration parameters which are text-based and fill the corresponding iDynTree object which is enum class-based for wrench types and iDyntree position and direction based.

I agree with what you guys suggested to update the class `iDynTree::UnknownWrenchContact` in iDynTree and to add the checks in `openContactFrames(config)` instead of the intermediate class.

Reply:
Thanks guys, in https://github.com/robotology/whole-body-estimators/pull/48/commits/edcd509dad868124ec98e627ecd260af425c8ca1, I removed this class and used `iDynTree::UnknownWrenchContact` instead. However, I am not sure it's needed to update `iDynTree::UnknownWrenchContact` anymore.